vignettes/datatable-importing.Rmd

```diff
@@ -195,6 +195,88 @@ For more canonical documentation of defining packages dependency check the offic
 
 Some of internally used C routines are now exported on C level thus can be used in R packages directly from their C code. See [`?cdt`](https://rdatatable.gitlab.io/data.table/reference/cdt.html) for details and [Writing R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html) _Linking to native routines in other packages_ section for usage.
 
-## Importing from non-R Applications {non-r-API}
+## Importing from non-r Applications {non-r-api}
 
 Some tiny parts of `data.table` C code were isolated from the R C API and can now be used from non-R applications by linking to .so / .dll files. More concrete details about this will be provided later; for now you can study the C code that was isolated from the R C API in [src/fread.c](https://github.com/Rdatatable/data.table/blob/master/src/fread.c) and [src/fwrite.c](https://github.com/Rdatatable/data.table/blob/master/src/fwrite.c).
+
+## How to convert your Depends dependency on data.table to Imports
+
+To convert a `Depends` dependency on `data.table` to an `Imports` dependency in an R package, follow these steps:
+
+### 1. Update the DESCRIPTION File
+In your package's `DESCRIPTION` file, change the entry from `Depends` to `Imports`. This involves moving `data.table` from the `Depends` field to the `Imports` field.
+
+**Before:**
+```r
+Depends:
+    R (>= 3.5.0),
+    data.table
+Imports:
+```
+
+**After:**
+```r
+Depends:
+    R (>= 3.5.0)
+Imports:
+    data.table
+```
+
+### 2. Modify the NAMESPACE File
+Ensure that any functions from `data.table` that you use in your package are imported. This can be done either by explicitly importing specific functions or by importing the entire package but specific imports are considered best practice.
```

Comment:
This section would frustrate me -- if I wrote a package with many files and 1000s of lines of code, how can I be sure I've imported _all_ the functions I've used? The description here sweeps the most difficult part of this step under the rug.

In the simple case of moving one package from Depends to Imports, I think the easiest solution is to just run `R CMD check` after step 1, _assuming your package is already passing `R CMD check` before starting. (So, let's make that "Step 0" -- Ensure your package is passing `R CMD check` initially).

Any symbols in your sources that are no longer defined will be identified by `R CMD check`.

Reply:
Note that this will also help with the other thing not called out by this section yet -- the need to import data.table's "special symbols" like `.N`, `.SD`, `:=`, etc. The same `R CMD check` approach will help there, but it deserves special flagging given how common this is in data.table code.